rgdps/usecases/messages.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import NamedTuple
+
+from rgdps import repositories
+from rgdps.common.context import Context
+from rgdps.constants.errors import ServiceError
+from rgdps.constants.users import UserPrivacySetting
+from rgdps.constants.users import UserRelationshipType
+from rgdps.models.message import Message
+from rgdps.models.user import User
+
+
+class MessageResponse(NamedTuple):
+    message: Message
+    user: User
+
+
+class PaginatedMessagesResponse(NamedTuple):
+    messages: list[MessageResponse]
+    total: int
+
+
+async def from_id(
+    ctx: Context,
+    user: User,
+    message_id: int,
+    include_deleted: bool = False,
+) -> MessageResponse | ServiceError:
+    message = await repositories.message.from_id(
+        ctx,
+        message_id=message_id,
+        include_deleted=include_deleted,
+    )
+
+    if message is None:
+        return ServiceError.MESSAGES_NOT_FOUND
+
+    if message.sender_user_id != user.id and message.recipient_user_id != user.id:
+        return ServiceError.MESSAGES_INVALID_OWNER
+
+    if message.sender_user_id == user.id:
+        recipient = await repositories.user.from_id(ctx, message.recipient_user_id)
+    else:
+        recipient = await repositories.user.from_id(ctx, message.sender_user_id)
+
+    if recipient is None:
+        return ServiceError.MESSAGES_INVALID_OWNER
+
+    return MessageResponse(message=message, user=recipient)
+
+
+async def from_recipient_user_id(
```

Comment:
I would robably name this something along the lines of `get_sent`

Reply:
resolved in 8109c21